### ISSUE TITLE
snetcfg.exe crashes on Win 8.1

### ISSUE THREAD

**Issue by Megachip** [vboxme/Portable-VirtualBox#6]
Name der fehlerhaften Anwendung: snetcfg_x64.exe, Version: 5.2.3790.0, Zeitstempel: 0x436618ee
Name des fehlerhaften Moduls: msvcrt.dll, Version: 7.0.9600.16384, Zeitstempel: 0x5215f944
Ausnahmecode: 0xc0000005
Fehleroffset: 0x00000000000026c4
ID des fehlerhaften Prozesses: 0x2090
Startzeit der fehlerhaften Anwendung: 0x01ced3eb3a9468bc
Pfad der fehlerhaften Anwendung: V:\Virtual Box\data\tools\snetcfg_x64.exe
Pfad des fehlerhaften Moduls: C:\WINDOWS\system32\msvcrt.dll
Berichtskennung: 7846f70a-3fde-11e3-b488-e0f847067f89
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist:

**Comment by p060477** [comment]
hi!
this BUG isn't solved yet is it..?
i've dl the 2010 visual c and also for x86..but the issue is still there...

pls reply!!

cheers!

:)

**Comment by p060477** [comment]
the BUG is still there
WHY..??

**Comment by jcazzie** [comment]
Hello everyone,

I have also been having this issue (running on windows 8.1). I can't say that I have had any luck in resolving it, but I have reached out to Runar Buvik, and he recommended that I post the following pictures that I documented when trying to analyze the error. I hope this might provide some insight into the nature of this issue. If anyone needs any more information, I'd be glad to provide it. 

![image](https://cloud.githubusercontent.com/assets/13002242/8284952/b86fde16-18c5-11e5-9218-eec306fa9b15.png)


After windows finishes, the following message pops up,
![image](https://cloud.githubusercontent.com/assets/13002242/8284958/c59a8ac8-18c5-11e5-89d7-5c9dbd6fdbb6.png)
When I try to debug this, I am met with the following dissassembly: 
![image](https://cloud.githubusercontent.com/assets/13002242/8284959/ccdbfb64-18c5-11e5-9352-40b9bb9212b9.png)
If I let this continue, instead of trying to debug, it will randomly either allow bridged networking to continue successfully (usually the first time I try to handle this error). All other times that I try it, it shows that there are no network adapters available. 
![image](https://cloud.githubusercontent.com/assets/13002242/8284963/d24877da-18c5-11e5-822c-e7487017fcbf.png)

Once again, if anyone can resolve this issue, I'd be immensely grateful, and if there's any information needed, I'd be glad to provide it.

Thanks,
Jcazzie

**Comment by p060477** [comment]
Hi Jcazzle!

first so many thxs indeed for yr post!
i'm diggin with this issue since april 2015...and i thought i was hopeless..

1) how you were able to contact Runar..??..and moreover how he replied to you..??
    i sent to him so many,UNREPLIED, emails...

2)i do not have yr 2° picture...i see is from Visual studio..i've have all visual installed..but
    i have not yr pic, i run win 8.1 64bit

3)so i have not also yr 3rd pic

4)but i have yr 4th one..EVERY first time the bridge net is available and working..BUT
  the second time no, as per yr useful pic,
  the third YES, the fourth NO,the fifth YES and so on and on..
  to sum up at the 1-3-5-7-9 ....tentatives =YES, and at 2-4-6-8......NO as per yr 4° pic

i really hope that some one should be able to solve it!!

a little workroud i found was to,with task scheduler, run portable vbox at win start
in that case the issue, and so the 1° pic, does not appear..
BUT in that case i'm not able to run it MINIMIZED, even if i add -minimized in the task scheduler

i always get the portable-vbox.exe window opened in the middle of my desktop
and then i gave to reduce it at icon MANUALLY by myself

hope this my experience helps someone more skill then me..

cheers!!

:)

**Comment by p060477** [comment]
Hi Jcazzle!
first so many thxs indeed for yr post!i'm diggin with this issue since april 2015...and i thought i was hopeless..
1) how you were able to contact Runar..??..and moreover how he replied to you..??    i sent to him so many,UNREPLIED, emails...
2)i do not have yr 2° picture...i see is from Visual studio..i've have all visual installed..but    i have not yr pic, i run win 8.1 64bit
3)so i have not also yr 3rd pic
4)but i have yr 4th one..EVERY first time the bridge net is available and working..BUT  the second time no, as per yr useful pic,  the third YES, the fourth NO,the fifth YES and so on and on..  to sum up at the 1-3-5-7-9 ....tentatives =YES, and at 2-4-6-8......NO as per yr 4° pic
i really hope that some one should be able to solve it!!
a little workroud i found was to,with task scheduler, run portable vbox at win startin that case the issue, and so the 1° pic, does not appear..BUT in that case i'm not able to run it MINIMIZED, even if i add -minimized in the task scheduler
i always get the portable-vbox.exe window opened in the middle of my desktopand then i gave to reduce it at icon MANUALLY by myself
hope this my experience helps someone more skill then me..
cheers!!
:)



Date: Mon, 22 Jun 2015 08:04:46 -0700

**Comment by p060477** [comment]
no one wants to reply..??..so so strange...
:(

**Comment by jcazzie** [comment]
Hello p060477, I'm so sorry that you've been working on this for so long! That's fascinating, though, I can't say that I've seen the same alternating behavior. Usually it just works the first time that I try the networking, and fails all the rest of the time. I'm also running windows 8.1. My impression is that Mr. Buvik must be extremely busy, so that may be why he isn't able to answer emails. If anyone could help solve this, that would be immensely appreciated!

**Comment by p060477** [comment]
hi!
Mr Runar should be so busy..BUT even 4 days ago was here..so WHY do not spend
a few time with the problems and issues of someone who is getting crazy since april 2015..??

are you able to start portable vbox reduced at icon..??

thxs so much..you are the only one so kind here!!

:)

**Comment by Megachip** [comment]
@jcazzie @p060477 Solution: Do not use Win 8.1 ;)

Runar also only adapted from the former programmer Michael. The problem isn't dedicated to the PVB Code itself, more to external libraries or binaries (looks like snetcfg isn't compatible with latest builds of windows and it isn't shipped anymore with later builds, afaik).
This is an open source project, so if you want to get a bug fixed, fix it and share your solution.
The maintain of this project can be only done in free time, and there isn't much of it :(

**Comment by p060477** [comment]
i followed yr solution  till my ten years old pc left me..
now i cannot...

is there a way to run PVB reduced at icon?
is there a way to have a simple reply here from Runar?

the mantra that something is free ever let me think that also a kick in the axs could be free..
but really not fine..at least afaik

cheers!

:)

**Comment by p060477** [comment]
runar ..is there anybody out there..??!!
if yes pls one hit..!!

cheers!!

:)

**Comment by runarbu** [comment]
@p060477 I am sorry that I have not been able to answer you directly on email. Unfortunately so do I get so meany email about Portable-VirtualBox that I am unable to keep up with the volume.

While I understand that this issue is very annoying to you, I must priority my time to fix the issues that is effecting the must users first, and your issue dos not effect so many users. I do however hope to I will get the time to look at it eventually.

**Comment by runarbu** [comment]
Looks like this snetcfg_x64.exe file comes from Windows 2003 server. Maybe Windows 8 ships with a newer, working one?

Have anyone tried to see if they have a snetcfg*.exe file on there local machine and use the local one to replace the one that comes with Portable-VirtualBox? To do so find it on your local machine and replace the standard one that we ship that is located in Portable-VirtualBoxs \data\tools\ folder.

![snetcfg_x64](https://cloud.githubusercontent.com/assets/1230798/8705428/e36c0e2a-2b29-11e5-8888-2482adc6c6d4.png)

**Comment by Zoli1972** [comment]
Hi,

If this works, then the app could look for the file on the local system, compare the versions, and if they differ, copy the file before going on with the startup procedure. If the file on the system is always located at the same place, maybe a link to it in the portable virtualbox directory could also help. The link could be created if the file has been found and if not, the app could try to link (or copy) it's own file from a certain location to the same place.

Zoli

**Comment by p060477** [comment]
Hi!

first so many thxs in adv for yr kind attention!

i really do not have that file on my win 8.1 64bit

i did a close search and the only copy is the one in :
 Portable-VirtualBoxs \data\tools\ folder

so how to fix this boring and heavy issue??...

pls reply !!

**Comment by p060477** [comment]
hi runar..really hope not to wait another nearly 70 days for yr appearing..!!
pls one hit in a timely manner...!!

i'm jockin..but not so much..is till april that this issue is there..

cheers!!

:)

**Comment by p060477** [comment]
Runar..Runaaar..
..o no..disappeared again...like the Canterville Ghost...??!!..
:(

**Comment by p060477** [comment]
hi..
just to let you know that the file you talk.
snetcfg.exe
is really NOT on my win 8.1 where there is this BUG-ISSUE
but moreover is really not also on my old win xp
where i really do not have this frustrating bug at all..

so the matter is really not in the file as you think...

any other useful ideas in the meantime..??

cheers!!

:)

**Comment by Megachip** [comment]
@p060477 snetcfg is included the PVB and will NOT work in Win 8(.1?) and up. It is not included in any Windows installation (per default). It belongs to the windows driver development kit and should be replaced with the sc command in for newer windows builds than Win 7.
Feel free to experiment with the sc command and provide a patch for pvb.

**Comment by p060477** [comment]
Hi!
finally a partially useful reply!
1)is not included also in my win xp..so WHY there is really NOT the BUG ??
2)if you are right this should be CLEARLY stated in the PVB project page,
  i really mean www.vbox.me
3)-IF- you think you are right PLEASE show here CLEARLY which are stpes by steps ALL the
  passages in order to finally have solved this increscious and frustrating BUG

if in the project page there is clearly written for istance how to set e net support:

Network support

To download of Portable-VirtualBox
Unpack from Portable-VirtualBox
Start from Portable-VirtualBox
Attitudes open (Tray –> attitudes, CTRL+5) –> rider Network (Tab) –> VirtualBox with network support start –> memory (save)
Terminate from Portable-VirtualBox
Start from Portable-VirtualBox
Driver installation agree
Wait
Selection of a VM and the network map to host interfaces stop
Attitudes make
FINISHED

the last word -FINISHED- is really WRONG...
so pls continue it and finish it finally to solve this very very strange BUG

hope you will be able also to give the solution instead of only put the problem under a light

we are all here waiting for yr PATCH

cheers!!!

:)

PS: RUNAR...where are you..??""
      pls beat a hit..

**Comment by p060477** [comment]
..now also megacghip disappear..??..

..do you think should be useful to install this..?'..:
https://www.microsoft.com/en-us/download/details.aspx?id=42273

pls reply in not a -era- time..

cheers!!

**Comment by p060477** [comment]
this is the scottish most populated place of ghosts..
is there anybody outthere..?'

pls hit a beat..

all dead..??!!

**Comment by Zoli1972** [comment]
The software is OpenSource, so maybe you sould know some things:

- there could be no support
- you can fix things by yourself, and share the solution with the
community. That's the sense of OpenSource software.
- if you want to force things, because you're in hurry, maybe it's better
you pay for some software, where you have support.
- Members here are doing their work *on their free time for NO money*.

Due to this, there is no reason to complain about missing members or
missing answers to your requests for long time. Please, consider that.
Maybe the reason why no one answers is simply because no one has a clue on
how to solve it (me too), or don't have the time looking for a solution (me
too). So please don't write further complaints here.

Feel free to look for a solution by yourself, and share it with others here.

Thanks

Z.

Zoltán Sebők

Tel. (DE) [+49] (0)89 420952369
Mobile (DE) [+49] (0)172 8262620
Tel. (HU) [+36] 21 3809901
Mobile (HU) [+36] 20 4235463
Homepage: https://sites.google.com/site/zoltanseboek

2015-07-26 10:19 GMT+02:00 p060477 <notifications@github.com>:

**Comment by p060477** [comment]
really hope someone catch yr:
" look for a solution by yourself, and share it with others here."

..at least Runar who has the responsibility of this portable vers
or MegaChip who through the stone but well -hide his hand ...



cheers!!

**Comment by p060477** [comment]
hi RUNAR!

hope you would like to be informed that with win 10 there is really NO way to use
net - bridge - at all..
here is the appcrash report:

Version=1
EventType=APPCRASH
EventTime=130829067204904507
ReportType=2
Consent=1
UploadTime=130829067205374946
ReportIdentifier=2ab288cc-384b-11e5-8532-74d435bb1f97
IntegratorReportIdentifier=bd092af0-1b5f-4b6a-ab3a-816ce36aec28
NsAppName=snetcfg_x64.exe
Response.BucketId=e40d3758771e7ddebbea60b856c3bea6
Response.BucketTable=4
Response.LegacyBucketId=120282220129
Response.type=4
Sig[0].Name=Nome applicazione
Sig[0].Value=snetcfg_x64.exe
Sig[1].Name=Versione applicazione
Sig[1].Value=5.2.3790.0
Sig[2].Name=Timestamp applicazione
Sig[2].Value=436618ee
Sig[3].Name=Nome modulo con errori
Sig[3].Value=msvcrt.dll
Sig[4].Name=Versione modulo con errori
Sig[4].Value=7.0.10240.16384
Sig[5].Name=Timestamp modulo con errori
Sig[5].Value=559f3b84
Sig[6].Name=Codice eccezione
Sig[6].Value=c0000005
Sig[7].Name=Offset eccezione
Sig[7].Value=000000000005594b
DynamicSig[1].Name=Versione SO
DynamicSig[1].Value=10.0.10240.2.0.0.768.101
DynamicSig[2].Name=ID impostazioni locali
DynamicSig[2].Value=1040
DynamicSig[22].Name=Informazioni aggiuntive 1
DynamicSig[22].Value=2b22
DynamicSig[23].Name=Ulteriori informazioni 2
DynamicSig[23].Value=2b22b632e70c9a1a86d8dc7127cf8bdd
DynamicSig[24].Name=Ulteriori informazioni 3
DynamicSig[24].Value=aa84
DynamicSig[25].Name=Ulteriori informazioni 4
DynamicSig[25].Value=aa849e5386a4c1641e958a78c1d95afb
UI[2]=G:\VIRTUALBOX 5.0.0-10.15.73\Portable-VirtualBox\data\tools\snetcfg_x64.exe
UI[3]=network config sample ha smesso di funzionare
UI[4]=Windows: è possibile ricercare online una soluzione al problema.
UI[5]=Cerca una soluzione online e chiudi il programma
UI[6]=Cerca una soluzione online in seguito e chiudi il programma
UI[7]=Chiudi il programma
LoadedModule[0]=G:\VIRTUALBOX 5.0.0-10.15.73\Portable-VirtualBox\data\tools\snetcfg_x64.exe
LoadedModule[1]=C:\WINDOWS\SYSTEM32\ntdll.dll
LoadedModule[2]=C:\WINDOWS\system32\KERNEL32.DLL
LoadedModule[3]=C:\WINDOWS\system32\KERNELBASE.dll
LoadedModule[4]=C:\WINDOWS\system32\apphelp.dll
LoadedModule[5]=C:\WINDOWS\AppPatch\AppPatch64\AcLayers.DLL
LoadedModule[6]=C:\WINDOWS\system32\msvcrt.dll
LoadedModule[7]=C:\WINDOWS\system32\USER32.dll
LoadedModule[8]=C:\WINDOWS\system32\GDI32.dll
LoadedModule[9]=C:\WINDOWS\system32\SHLWAPI.dll
LoadedModule[10]=C:\WINDOWS\system32\combase.dll
LoadedModule[11]=C:\WINDOWS\system32\RPCRT4.dll
LoadedModule[12]=C:\WINDOWS\SYSTEM32\sfc.dll
LoadedModule[13]=C:\WINDOWS\SYSTEM32\WINSPOOL.DRV
LoadedModule[14]=C:\WINDOWS\SYSTEM32\bcrypt.dll
LoadedModule[15]=C:\WINDOWS\SYSTEM32\sfc_os.DLL
LoadedModule[16]=C:\WINDOWS\SYSTEM32\SortWindows61.dll
LoadedModule[17]=C:\WINDOWS\system32\IMM32.DLL
LoadedModule[18]=C:\WINDOWS\system32\MSCTF.dll
LoadedModule[19]=C:\WINDOWS\system32\ole32.dll
LoadedModule[20]=C:\WINDOWS\system32\sechost.dll
LoadedModule[21]=C:\WINDOWS\system32\SETUPAPI.dll
LoadedModule[22]=C:\WINDOWS\system32\CFGMGR32.dll
LoadedModule[23]=C:\WINDOWS\system32\kernel.appcore.dll
LoadedModule[24]=C:\WINDOWS\SYSTEM32\bcryptPrimitives.dll
LoadedModule[25]=C:\WINDOWS\system32\uxtheme.dll
LoadedModule[26]=C:\WINDOWS\system32\clbcatq.dll
LoadedModule[27]=C:\Windows\System32\NetSetupShim.dll
LoadedModule[28]=C:\WINDOWS\system32\WS2_32.dll
LoadedModule[29]=C:\WINDOWS\system32\NSI.dll
LoadedModule[30]=C:\Windows\System32\NetSetupApi.dll
State[0].Key=Transport.DoneStage1
State[0].Value=1
FriendlyEventName=Ha smesso di funzionare
ConsentKey=APPCRASH
AppName=network config sample
AppPath=G:\VIRTUALBOX 5.0.0-10.15.73\Portable-VirtualBox\data\tools\snetcfg_x64.exe
NsPartner=windows
NsGroup=windows8
ApplicationIdentity=ED3FA8EECD0FAC1001B44658EE510D0F


any ideas...??!!

**Comment by p060477** [comment]
runar..have you finally manage this very big issue..??

**Comment by abhishekhp** [comment]
Any updates on this ?

**Comment by p060477** [comment]
ask to Runar or Megachip
..if they are still alive..
what a shame..
:(

**Comment by Solice55** [comment]
This thread was painful to read... In any case, in the spirit of open source solutions, combined with Google and some problem solving skills, I present the following stream of consciousness as a workaround for the current problem.

Host OS: Windows 10 Pro 64-bit
Virtualbox Version: 5.0.22
Portable-Virtualbox Version: 6.4.9.0

The main problem here is snetcfg*.exe is not able to do its job to install all of the required network drivers and services.  Assuming that the files exist for said drivers, we may be able to invoke them manually.  This may make our portable Virtualbox environment somewhat less portable, but at least it will work.

Preliminary steps to this procedure:
- Make sure to have Portable-Virtualbox extracted and otherwise working, so that all of the files it uses are in the filesystem.
- Make sure that a non-portable install of Virtualbox is NOT installed.  This seems to make Portable-Virtualbox quit out and not set all the things it needs to set in order to function as expected.
- Make sure that you have Portable-Virtualbox set to NOT load networking drivers.
- Make sure that all Windows 10 Hyper-V components are not installed; they tend to muck with Virtualbox in general. (Start -> search for "turn windows features on or off", uncheck Hyper-V categories)

Having done all of that, follow this procedure to get a bridged adapter available to your portable VM's:
- Navigate to the Network Connections Control Panel, and right-click the interface you want to use and click Properties.  (This procedure will address all interfaces that are currently installed)
- On the following dialog box, click Install..., then Service, then Have Disk.
- Navigate to a sub-directory of your Portable-Virtualbox directory: [...]\Portable-VirtualBox\app64\drivers\network
- Here, you will notice that there are several sub-directories.  You will need to try more than one, but for me, I had to add the driver that was under netlwf.  This installed a persistent Virtualbox bridge driver for each interface that you have installed, not just the one selected.
- Start up Portable-Virtualbox.  Do not enable the autoload network drivers option.
- If you need other interfaces or services, follow a similar procedure to install them.

To the project maintainer: I recommend finding another way to auto-install network services; the snetcfg*.exe does not appear to play nice with Windows 10.

Good luck.

**Comment by p060477** [comment]
Hi Solice!
finally some useful stuff!
thxs so much!
here runar has desappeared as the canterville ghost...:(
anyway let's go to the main point:
yr intriguing trick works also for having portable vb with network adap -bridged-?
i ask cause i use emule and that is the only way to not have it firewalled and with the 2 arrows well green!
thxs so much indeed!
cheers!

Runar ..have you read this..??..and Megachip..??..

**Comment by Solice55** [comment]
I used this workaround specifically to use Portable-Virtualbox with a bridged network adapter.  As i noted, you may find that it works with other Virtualbox-specific drivers if you simply install them.  You will have to try for yourself.  For example, if you needed the virtual-machine-only adapter, you could simply install it with one of the drivers provided in that network directory.

**Comment by Peppernrino** [comment]
to add to the conversation a bit, it seems runar was attempting to find a workaround using sc. i found this stackoverflow thread from november 2015: https://stackoverflow.com/questions/33618303/installing-inf-file-in-windows-10-migrating-away-from-snetcfg-exe

perhaps someone can expand upon what is being tried here?

i also saw someone mentioning somewhere running it in windows 8.1 compatibility mode... i wonder if there's a way to force that? i intend to package the final result for portableapps.com... maybe there is some way to force that option in the .ini files once i get it to that point... just thinking out loud. maybe somebody can take something from it. :)

**Comment by Peppernrino** [comment]
perhaps this is of use as well? : https://stackoverflow.com/questions/31922055/bridged-networking-not-working-in-virtualbox-under-windows-10

**Comment by p060477** [comment]
it should be very very a -miracle- to have bridge net working in vb portable with win 10...runar has disappeared since years...
if someone find a solution to this very very -cold case- pls post it here..i pray...!!..
cheers :)

**Comment by Peppernrino** [comment]
i think the intention many years ago was for me to take over this repo with runar to some degree... maybe i can figure it out. i just got back into this. haven't looked at it since about 2016. and i'm sorry. lol

**Comment by p060477** [comment]
since the infaust issue of win 10 this is the most -unsolved- bug for vb portable..and also its vbox.me project...
i hope to see it finally solved before my departure on this land.. ;)  :)

**Comment by Peppernrino** [comment]
there might also be something to this: https://www.minitool.com/news/start-service-failed-1058.html

**Comment by p060477** [comment]
...oh..if it was so simple...it should have been fixed in a while...i think also the -disappeared runar- should have been able too...
;)

**Comment by Peppernrino** [comment]
update: i have made many forms of progress. i have never used autoit, but i don't think that is the final solution... i am looking into employing the use of other layers during another kind of install... being secretive because i want to be the one that does it... lol. i have gathered a list of registry entries etc, and might be able to slip things in and out with some magic... stay tuned, friend. :)

**Comment by p060477** [comment]
hi! ...i maintain my fingers closely crossed...and i pray for yr efforts to be finally usefull and working..!!
don't give up!
;)

**Comment by Peppernrino** [comment]
also, i understand your frustrations, and no offense, but i have noticed you kind of ripping on people when they aren't able to produce the desired final result. open source development is basically volunteer work. just thank them for trying, and hope their progress helps the next person. it's an imperfect world. we have to accept that not everyone can do every thing. you could be driving away someone that has a great idea somewhere down the road. :)

for example, i would not be as far as i am if it weren't for @timsky and their contributions. their rewrite of the autoit stuff is slick. it's missing some small things... changing flags for install version, etc, but i have fixed those, and will submit a change when i get everything working properly. i'm sure you can see there is already only like.... 3 people in the entire world working on this. lol. let's try to encourage a more positive atmosphere, even in light of basically constant failure. haha. it will happen. we just need the right people on it. don't worry. <3

**Comment by p060477** [comment]
i totally agree with you!
i really hope you will finally solve this very old vbox portable bug!
keep goin on!
cheers!
LoL :) !
